### PR TITLE
Send unregister via AMQP

### LINKED ISSUE #11: Send 'unregister' message to message queue
- [ ] Replace msg_unregister to use rabbitmq
- [ ] Publish to device.unregister
- [ ] Replace proxy_removed by device_removed callback

### PR DESCRIPTION
**Pull request by ramonhpr**
This PR closes #11 

How to test:
install pika python package
run: `test/mock-connector.py`
run knotd: `sudo src/knotd -nr -c src/knotd.conf`
register a thing with thing mocked script: `test/test-conn.py --debug`
Use d-feet to remove the device.

Expected result:
the mock-connector should show the messages in json
the test-conn.py should receive unregister request and show
```
DEBUG:root:[payload_len: 0] receive knot_msg = 0x12 (UNREGISTER_REQUEST)
INFO:root:Unregistered
INFO:root:Closing connection
```
And knotd should remove the dbus device in d-feet

### CODE REVIEW

File: test/mock-connector.py [hunk 1]
```diff
@@ -54,6 +58,8 @@ def callback(ch, method, properties, body):
         message = body
         channel.basic_publish(exchange=fog_exchange,
                               routing_key=KEY_UNREGISTERED, body=message)
+    elif method.routing_key == EVENT_DATA:
+        return
```
Comment (alla3) [inline test/mock-connector.py]: Please `return None`.

File: src/cloud.c [hunk 2]
```diff
@@ -175,25 +182,13 @@ int cloud_set_cbs(cloud_downstream_cb_t on_update,
...
-					   AMQP_EVENT_DEVICE_UNREGISTERED);
-	if (err) {
-		hal_log_error("Error on set up queue to consume.\n");
-		return -1;
+	for (i = 0; fog_events[i] != NULL; i++) {
+		err = amqp_set_queue_to_consume(queue_fog, AMQP_EXCHANGE_FOG,
+					fog_events[i]);
+		if (err) {
+			hal_log_error("Error on set up queue to consume.\n");
+			return -1;
```
Comment (alla3) [inline src/cloud.c]: Shouldn't you call `amqp_bytes_free(queue_fog);` in case of failure?

File: test/mock-connector.py (hunk 1 above)
Comment (ramonhpr) [reply test/mock-connector.py]: Done

File: src/cloud.c (hunk 2 above)
Comment (ramonhpr) [reply src/cloud.c]: Done

File: src/cloud.h [hunk 3]
```diff
@@ -22,12 +22,17 @@
 typedef bool (*cloud_downstream_cb_t) (const char *id, struct l_queue *data,
 				       void *user_data);
 
+typedef void (*cloud_device_removed_cb_t) (const char *device_id,
```
Comment (vitbaq) [inline src/cloud.h]: What do you think about to use just id instead of device_id? To keep the pattern of cloud_downstream_cb_t.

File: src/msg.c [hunk 4]
```diff
@@ -1329,7 +1333,7 @@ static void unregister_callback(struct l_timeout *timeout, void *user_data)
 	device_forget_destroy(mydevice);
 }
 
-static void proxy_removed(const char *device_id, void *user_data)
+static void on_device_removed(const char *device_id, void *user_data)
```
Comment (vitbaq) [inline src/msg.c]: Do we have any case that needs a function return?
Such as adding the msg on the queue again.

File: src/cloud.h (hunk 3 above)
Comment (alla3) [reply src/cloud.h]: I think device_id is more clear than id, maybe we should standardize the id as device_id.

Comment (vitbaq) [reply src/cloud.h]: I agree. But will be out of the pattern. So I think that we can change to device_id after on another commit.

Comment (alla3) [reply src/cloud.h]: Indeed.

File: src/msg.c (hunk 4 above)
Comment (ramonhpr) [reply src/msg.c]: No. The thing should be removed even if it isn't connected.

Comment (vitbaq) [reply src/msg.c]: Ok

File: src/cloud.h (hunk 3 above)
Comment (ramonhpr) [reply src/cloud.h]: Done

**Pull request merged**